lib/_stream_writable.js

```diff
@@ -520,23 +529,37 @@ function needFinish(state) {
           !state.finished &&
           !state.writing);
 }
-
-function prefinish(stream, state) {
-  if (!state.prefinished) {
+function callFinal(stream, state) {
+  stream._final((err) => {
+    state.pendingcb--;
+    if (err) {
+      stream.emit('error', err);
+    }
     state.prefinished = true;
     stream.emit('prefinish');
+    finishMaybe(stream, state);
+  });
+}
+function prefinish(stream, state) {
+  if (!state.prefinished && !state.finalCalled) {
+    if (typeof stream._final === 'function') {
+      state.pendingcb++;
+      state.finalCalled = true;
+      process.nextTick(callFinal, stream, state);
```

Comment:
Hmm... Is `process.nextTick` really needed here? Can someone confirm/deny?

Reply:
@szmarczak This looks like a classic case where `process.nextTick()` is called for-- it's a case where the final function may be sync or async. This post explains more about it. https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/#process-nexttick

Reply:
I know what `process.nextTick` does :P Well, in my case, `_final()` is sync and `process.nextTick()` is useless. I haven't checked if that's required by some other Node code, so I'm just asking :)